weed/storage/needle_map_leveldb.go

```diff
@@ -330,5 +327,17 @@ func (m *LevelDbNeedleMap) DoOffsetLoading(v *Volume, indexFile *os.File, startF
 	} else {
 		return NeedleMapMetricFromIndexFile(indexFile, &mm, startFrom)
 	}
+
+	//set watermark
+	stat, err := indexFile.Stat()
+	if err != nil {
+		return err
+	}
+	watermark := (stat.Size() / types.NeedleMapEntrySize / watermarkBatchSize) * watermarkBatchSize
```

Comment:
seems this bug was not caught by the tests. Can you add tests to address this?

Reply:
Sure, I'll add the test case for this bug.

Reply:
Done

Reply:
Another question, is it ok to use reflect in test case?

Reply:
Now it's ready for review.
The statistics methord for file count is different between memory and leveldb index, which is confusing.
If use leveldb, entry count(idxSize/entrySize) = file count + deleted count.
If use memory, the above equationis is not  satisfied.